### PR TITLE
feat: add `Reducible.{exists, forall, foreach, findLeft, findRight, memberOf}`

### PR DESCRIPTION
**Pull request by jaschdoc**
Related to #1678

### CODE REVIEW

**Comment by magnus-madsen** [comment]
RFR?

**Comment by jaschdoc** [comment]
Yes!

File: main/src/library/Reducible.flix [hunk 1]
```diff
@@ -150,39 +150,49 @@ pub lawless class Reducible[t: Type -> Type] {
...
-    pub def find(f: a -> Bool & ef, t: t[a]): Option[a] = ???
+    pub def find(f: a -> Bool & ef, t: t[a]): Option[a] & ef =
+        Reducible.findLeft(f, t)
 
     ///
     /// Optionally returns the first element of `t` that satisfies the predicate `f` when searching from left to right.
     ///
-    pub def findLeft(f: a -> Bool & ef, t: t[a]): Option[a] = ???
+    pub def findLeft(f: a -> Bool & ef, t: t[a]): Option[a] & ef =
+        Reducible.foldLeft((acc, a) -> match acc {
```
Comment (magnus-madsen) [inline main/src/library/Reducible.flix:180]: Option.withDefault

**Comment by jaschdoc** [comment]
RFR

**Comment by jaschdoc** [comment]
Added tests that verify that elements are visited in the correct order when the functions is specified as either left or right associative

**Comment by magnus-madsen** [comment]
Neat! Thanks

**Pull request merged**